.github/workflows/run_mmperf.yml

```
@@ -0,0 +1,106 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# Workflow for running `mmperf` (https://github.com/mmperf/mmperf).
+# `mmperf` benchmarks matrix-multiply workloads on IREE and other backends such
+# as RUY, TVM, Halide, CuBLAS, etc.
+#
+# The workflow runs benchmarks on CPU and GPU and uploads results to the
+# `mmperf-benchmark-artifacts` GC bucket.
+
+name: mmperf
+
+on:
+  workflow_dispatch:
+  pull_request:
+
+jobs:
+  setup:
+    runs-on: ubuntu-20.04
+    env:
+      # The commit being checked out is the merge commit for the PR. Its first
+      # parent will be the tip of main.
+      BASE_REF: HEAD^
+      PR_TITLE: ${{ github.event.pull_request.title }}
+      PR_BODY: ${{ github.event.pull_request.body }}
+    outputs:
+      should-run: ${{ steps.configure.outputs.should-run }}
+      ci-stage: ${{ steps.configure.outputs.ci-stage }}
+      runner-env: ${{ steps.configure.outputs.runner-env }}
+      runner-group: ${{ steps.configure.outputs.runner-group }}
+      write-caches: ${{ steps.configure.outputs.write-caches }}
+    steps:
+      - name: "Checking out repository"
+        uses: actions/checkout@7884fcad6b5d53d10323aee724dc68d8b9096a2e # v2
+        with:
+          # We need the parent commit to do a diff
+          fetch-depth: 2
+      - name: "Configuring CI options"
+        id: configure
+        run: |
+          # Just informative logging. There should only be two commits in the
+          # history here, but limiting the depth helps when copying from a local
+          # repo instead of using checkout, e.g. with
+          # https://github.com/nektos/act where there will be more.
+          git log --oneline --graph --max-count=3
+          ./build_tools/github_actions/configure_ci.py
+
+  benchmark_cpu:
+    needs: setup
+    if: needs.setup.outputs.should-run == 'true'
+    runs-on:
+      - self-hosted  # must come first
+      - runner-group=${{ needs.setup.outputs.runner-group }}
+      - environment=${{ needs.setup.outputs.runner-env }}
+      - cpu
+      - os-family=Linux
+    env:
+      MMPERF_RESULTS_DIR: mmperf-results
+      GCS_UPLOAD_DIR: gs://mmperf-benchmark-artifacts
+    steps:
+      - name: "Checking out repository"
+        uses: actions/checkout@7884fcad6b5d53d10323aee724dc68d8b9096a2e # v2
+        with:
+          submodules: true
+      - name: "Running mmperf on CPU"
+        run: |
+          mkdir ${MMPERF_RESULTS_DIR}
+          ./build_tools/github_actions/docker_run.sh \
+            gcr.io/iree-oss/mmperf@sha256:e563ead2db6f7c28a4fb815fe14444d38e40c49ba5b37b0d6dba741825570632 \
+          ./build_tools/benchmarks/mmperf/run_mmperf.sh "${MMPERF_RESULTS_DIR}" "cpu"
+      - name: "Uploading artifacts"
+        run: |
+          export DIR_NAME=`ls -lat ${MMPERF_RESULTS_DIR}/latest | awk '{print $NF}'`
```

Parsing the output of `ls` or using it programatically is generally discouraged. The output is intended for human, not machine, consumption. It took me a while to parse what you were trying to get here. I guess `latest` is a symlink that points to another directory in the same parent directory? There are proper commands for that. If what you want is the literal value of the symbolic link, use `readlink`. If you want to resolve the path of what it's pointing to, use `readlink -f` or `realpath`. If you want its path relative to some other directory (e.g. the mmperf directory), use `realpath --relative-to /foo/bar`

Also, don't use $() instead of backticks for command substitution (https://google.github.io/styleguide/shellguide.html#command-substitution)

```suggestion
          export DIR_NAME="$(readlink "${MMPERF_RESULTS_DIR}/latest)"
```

---

Thanks for the tips.